AVInput/AVInput.cpp

```diff
@@ -224,5 +327,923 @@ string AVInput::currentVideoMode(bool &success)
+    }
+}
+
+void AVInput::dsAVVideoModeEventHandler(const char *owner, IARM_EventId_t eventId, void *data, size_t len)
+{
+    if(!AVInput::_instance)
+        return;
+
+    if (IARM_BUS_DSMGR_EVENT_HDMI_IN_VIDEO_MODE_UPDATE == eventId) {
+        IARM_Bus_DSMgr_EventData_t *eventData = (IARM_Bus_DSMgr_EventData_t *)data;
+        int hdmi_in_port = eventData->data.hdmi_in_video_mode.port;
+        dsVideoPortResolution_t resolution;
+        resolution.pixelResolution =  eventData->data.hdmi_in_video_mode.resolution.pixelResolution;
+        resolution.interlaced =  eventData->data.hdmi_in_video_mode.resolution.interlaced;
+        resolution.frameRate =  eventData->data.hdmi_in_video_mode.resolution.frameRate;
+        LOGWARN("Received IARM_BUS_DSMGR_EVENT_HDMI_IN_VIDEO_MODE_UPDATE  event  port: %d, pixelResolution: %d, interlaced : %d, frameRate: %d \n", hdmi_in_port,resolution.pixelResolution, resolution.interlaced, resolution.frameRate);
+        AVInput::_instance->AVInputVideoModeUpdate(hdmi_in_port, resolution);
+    }
+}
+
+void AVInput::dsAVGameFeatureStatusEventHandler(const char *owner, IARM_EventId_t eventId, void *data, size_t len)
+{
+    if(!AVInput::_instance)
+        return;
+
+    if (IARM_BUS_DSMGR_EVENT_HDMI_IN_ALLM_STATUS == eventId)
+    {
+        IARM_Bus_DSMgr_EventData_t *eventData = (IARM_Bus_DSMgr_EventData_t *)data;
+        int hdmi_in_port = eventData->data.hdmi_in_allm_mode.port;
+        bool allm_mode = eventData->data.hdmi_in_allm_mode.allm_mode;
+        LOGWARN("Received IARM_BUS_DSMGR_EVENT_HDMI_IN_ALLM_STATUS  event  port: %d, ALLM Mode: %d", hdmi_in_port,allm_mode);
+
+        AVInput::_instance->AVInputALLMChange(hdmi_in_port, allm_mode);
+    }
+}
+
+void AVInput::AVInputALLMChange( int port , bool allm_mode)
+{
+    JsonObject params;
+    params["id"] = port;
+    params["gameFeature"] = "ALLM";
+    params["mode"] = allm_mode;
+
+    sendNotify(AVINPUT_EVENT_ON_GAME_FEATURE_STATUS_CHANGED, params);
+    GetHandler(2)->Notify(AVINPUT_EVENT_ON_GAME_FEATURE_STATUS_CHANGED, params);
+}
+
+uint32_t AVInput::getSupportedGameFeatures(const JsonObject& parameters, JsonObject& response)
+{
+    LOGINFOMETHOD();
+    vector<string> supportedFeatures;
+    try
+    {
+        device::HdmiInput::getInstance().getSupportedGameFeatures (supportedFeatures);
+        for (size_t i = 0; i < supportedFeatures.size(); i++)
+        {
+            LOGINFO("Supported Game Feature [%zu]:  %s\n",i,supportedFeatures.at(i).c_str());
+        }
+    }
+    catch (const device::Exception& err)
+    {
+        LOG_DEVICE_EXCEPTION0();
+    }
+
+    if (supportedFeatures.empty()) {
+        return Core::ERROR_GENERAL;
+    }
+    else {
+        setResponseArray(response, "suppoortedGameFeatures", supportedFeatures);
+        return Core::ERROR_NONE;
+    }
+}
+
+uint32_t AVInput::getGameFeatureStatusWrapper(const JsonObject& parameters, JsonObject& response)
+{
+    string sPortId = parameters["portId"].String();
```

Comment:
please use `parameters["portId"].Number()` if it is number

Reply:
Modified code to use Number()